src/WebJobs.Script.WebHost/WebScriptHostManager.cs

```diff
@@ -289,7 +291,25 @@ public FunctionDescriptor GetHttpFunctionOrNull(HttpRequestMessage request)
             }
 
             FunctionDescriptor function = null;
-            var routeData = _httpRoutes.GetRouteData(request);
+
+            // if true, the match will first be against local http trigger functions otherwise it will be against proxies and functions combined with proxies matching first which is the default behavior.
+            if (functionRoutesFirst)
+            {
+                function = GetFunctionDescriptorFromRoute(request, _httpFunctionsOnlyRoutes.GetRouteData(request));
+            }
+
+            if (function == null)
+            {
+                function = GetFunctionDescriptorFromRoute(request, _httpRoutes.GetRouteData(request));
```

Comment:
This double lookup scheme will mean we attempt to match against all function routes twice, in cases where functionRoutesFirst == true.

I wonder if it wouldn't be simpler to have a single route collection, and add a discriminator to IHttpRoute.DataTokens for each proxy. That would allow us to do filtered queries on the route collection, e.g. via an extension method something like:

```csharp
public static class HttpRouteExtensions
{
    public static IHttpRouteData GetRouteData(this HttpRouteCollection routes, HttpRequestMessage request, bool functionRoutesFirst = false)
    {
        IEnumerable<IHttpRoute> orderedRoutes = routes;
        if (functionRoutesFirst)
        {
            orderedRoutes = routes.OrderBy(p => !(bool)p.DataTokens["IsProxyRoute"]);
        }

        return orderedRoutes.Select(p => p.GetRouteData(routes.VirtualPathRoot, request)).FirstOrDefault();
    }
}
```

Reply:
Pull request here: https://github.com/Azure/azure-webjobs-sdk-script/pull/2222